test/integration/cluster_dns_test.go

```diff
@@ -0,0 +1,64 @@
+// +build integration
+
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	commonutil "k8s.io/minikube/pkg/util"
+	"k8s.io/minikube/test/integration/util"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestClusterDNS(t *testing.T) {
+	minikubeRunner := util.MinikubeRunner{BinaryPath: *binaryPath, T: t}
+	dnsTest := func() error {
+		if minikubeRunner.GetStatus() != "Running" { //make/use status constants?
+			minikubeRunner.RunCommand("start", true)
+		}
+		minikubeRunner.CheckStatus("Running")
+
+		kubectlRunner := util.NewKubectlRunner(t)
+		podName := "busybox"                                 //make/use a constant?
+		podPath, _ := filepath.Abs("test_data/busybox.yaml") //make/use a constant?
+
+		const strLen = 20 //ask Dan about this value?
+		podNamespace := kubectlRunner.CreateRandomNamespace(strLen)
+
+		kubectlRunner.RunCommandSimple([]string{"create", "-f", podPath, "--namespace=" + podNamespace})
+		defer kubectlRunner.RunCommandSimple([]string{"delete", "namespace", podNamespace})
+
+		defer kubectlRunner.RunCommandSimple([]string{"delete", "-f", podPath, "--namespace=" + podNamespace})
+
+		podStatus := kubectlRunner.RunCommandSimple([]string{"get", "pod", "--namespace=" + podNamespace})
```

Comment:
See my test in addons_test. You can use RunCommand and parse the output into a real PodStatus struct. It's a little less fragile than string comparison.